### PR TITLE
Thread Mid-Execution capture through Gapit.

### PR DESCRIPTION
**Pull request by AWoloszyn**

### CODE REVIEW

File: gapii/cc/connection_header.h [hunk 1]
```diff
@@ -48,6 +50,8 @@ class ConnectionHeader {
     uint32_t mVersion;                      // 2 or 3
     uint32_t mObserveFrameFrequency;        // non-zero == enabled. Version: 2+
     uint32_t mObserveDrawFrequency;         // non-zero == enabled. Version: 2+
+    uint32_t mStartFrame;                   // non-zero == Frame to start at. version 4+
+    uint32_t mNumFrames;                    // non-zero == Number of frames to capture. version 4+
```
Comment (Qining) [inline gapii/cc/connection_header.h]: Just wonder if we should put the `mStartFrame` and `mNumFrames` to the end of the header struct?

File: gapii/cc/spy.cpp [hunk 2]
```diff
@@ -186,6 +189,11 @@ Spy::Spy()
                 (header.mFlags & ConnectionHeader::FLAG_DISABLE_PRECOMPILED_SHADERS) != 0;
         mRecordGLErrorState =
                 (header.mFlags & ConnectionHeader::FLAG_RECORD_ERROR_STATE) != 0;
+        // This will be over-written if we also set the header flags
+        mSuspendCaptureFrames = header.mStartFrame;
+        mCaptureFrames = header.mNumFrames;
+        mSuspendCaptureFrames.store((header.mFlags & ConnectionHeader::FLAG_DEFER_START)?
+            -1: mSuspendCaptureFrames.load());
```
Comment (Qining) [inline gapii/cc/spy.cpp]: Maybe make the `-1` a constant defined somewhere with an easy to understand name?

File: gapii/cc/connection_header.h (hunk 1 above)
Comment (AWoloszyn) [inline gapii/cc/connection_header.h]: I wanted to leave the "numbers" with the other "numbers" in the header. In this case mFlags is a bit-field.

File: gapii/cc/spy.cpp (hunk 2 above)
Comment (AWoloszyn) [inline gapii/cc/spy.cpp]: Done

**Pull request merged**